### PR TITLE
Add an option to prevent Worker from catching exceptions from Task.run()

### PR DESCRIPTION
**Pull request by belevtsoff**
Addresses #1201 

Here I used the fact that `multiprocessing.Queue` does basic serialization itself, which should work well for exceptions. I also wanted to create a container class TaskProcessResult to formalize a bit the growing set of attributes that is passed over the queue and read in several places, but I have no experience with mock and didn't know how to adjust the tests. Do you think that would be an overkill?

### CODE REVIEW

**Comment by erikbern** [comment]
LGTM

File: luigi/worker.py [hunk 1]
```diff
@@ -700,6 +709,10 @@ def _handle_next_task(self):
                 continue
             self._running_tasks.pop(task_id)
 
+            # raise exception if needed
```
Comment (Tarrasch) [inline luigi/worker.py]: I would just remove this comment. The code is definitely readable enough. :)

**Comment by Tarrasch** [comment]
LGTM. May I just ask that you merge the commits?

**Comment by belevtsoff** [comment]
@Tarrasch do you mean you'd like to have a single commit for the whole pull request?

File: luigi/worker.py (hunk 1 above)
Comment (belevtsoff) [inline luigi/worker.py]: fair enough :)

**Comment by Tarrasch** [comment]
@belevtsoff Yea, I think that keeps the commit log clener. And when you git blame the test lines, you'll also see the implementation that came with it, and vice verse.

**Comment by belevtsoff** [comment]
@Tarrasch done

**Comment by Tarrasch** [comment]
Thanks @belevtsoff!

**Pull request merged**